core/app/models/spree/promotion.rb

```diff
@@ -26,31 +31,45 @@ class Promotion < Spree::Base
 
     before_save :normalize_blank_values
 
-    scope :coupons, ->{ where("#{table_name}.code IS NOT NULL") }
-
     order_join_table = reflect_on_association(:orders).join_table
-
-    scope :applied, -> { joins("INNER JOIN #{order_join_table} ON #{order_join_table}.promotion_id = #{table_name}.id").uniq }
+    scope :applied, -> do
+      joins("INNER JOIN #{order_join_table} ON " \
+        "#{order_join_table}.promotion_id = #{table_name}.id").distinct
+    end
+    scope :coupons, -> do
+      joins(:promotion_codes).where("#{Spree::PromotionCode.table_name}.value IS NOT NULL")
```

Comment:
A code without a value doesn't make any sense. I would prefer that `Spree::PromotionCode` would validate that value is present, not null, and valid in all other important ways. Then drop this `where` condition.